frontend/src/app/main/data/workspace/variants.cljs

```diff
@@ -0,0 +1,229 @@
+            data    (dsh/lookup-file-data state)
+
+            related-components (->> (:components data)
+                                    vals
+                                    (filter #(= (:variant-id %) variant-id)))
+
+            changes (-> (pcb/empty-changes it page-id)
+                        (pcb/with-library-data data))
+
+            changes (reduce (fn [changes component]
+                              (pcb/update-component
+                               changes (:id component)
+                               #(assoc-in % [:variant-properties pos :name] new-name)))
+                            changes
+                            related-components)
+            undo-id (js/Symbol)]
+        (rx/of
+         (dwu/start-undo-transaction undo-id)
+         (dch/commit-changes changes)
+         (dwu/commit-undo-transaction undo-id))))))
+
+
+
+
+(defn remove-property
+  "Remove the variant property on the position pos
+   in all the components with this variant-id"
+  [variant-id pos]
+  (ptk/reify ::remove-property
+    ptk/WatchEvent
+    (watch [it state _]
+      (let [page-id (:current-page-id state)
+            data    (dsh/lookup-file-data state)
+            objects    (dsh/lookup-page-objects state page-id)
+
+            related-components (->> (:components data)
+                                    vals
+                                    (filter #(= (:variant-id %) variant-id)))
+
+            changes (-> (pcb/empty-changes it page-id)
+                        (pcb/with-library-data data)
+                        (pcb/with-objects objects))
+
+            changes (reduce (fn [changes component]
+                              (let [props   (:variant-properties component)
+                                    props   (vec (concat (subvec props 0 pos) (subvec props (inc pos))))
+                                    main-id (:main-instance-id component)
+                                    name    (properties-to-name props)]
+                                (-> changes
+                                    (pcb/update-component (:id component) #(assoc % :variant-properties props))
+                                    (pcb/update-shapes [main-id] #(assoc % :variant-name name)))))
+                            changes
+                            related-components)
+            undo-id (js/Symbol)]
+        (rx/of
+         (dwu/start-undo-transaction undo-id)
+         (dch/commit-changes changes)
+         (dwu/commit-undo-transaction undo-id))))))
+
+
+(defn update-property-value
+  "Updates the variant property value on the position pos in a component"
+  [component-id pos value]
+  (ptk/reify ::update-property-value
+    ptk/WatchEvent
+    (watch [it state _]
+      (let [page-id   (:current-page-id state)
+            data      (dsh/lookup-file-data state)
+            objects   (dsh/lookup-page-objects state page-id)
+            component (dm/get-in data [:components component-id])
```

Comment:
I think `app.common.types.component-list/get-component` helper should be used here